multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -187,129 +207,136 @@ test.serial('lp deposits', async t => {
   );
 });
 
-test.serial('advance and settlement', async t => {
-  const {
-    nobleTools,
-    nobleAgoricChannelId,
-    oracleWds,
-    retryUntilCondition,
-    useChain,
-    usdcOnOsmosis,
-    vstorageClient,
-  } = t.context;
-
-  // EUD wallet on osmosis
-  const eudWallet = await createWallet(useChain('osmosis').chain.bech32_prefix);
-  const EUD = (await eudWallet.getAccounts())[0].address;
-
-  // parameterize agoric address
-  const { settlementAccount } = await vstorageClient.queryData(
-    `published.${contractName}`,
-  );
-  t.log('settlementAccount address', settlementAccount);
+const advanceAndSettleScenario = test.macro({
+  title: (_, mintAmt: bigint, eudChain: string) =>
+    `advance ${mintAmt} uusdc to ${eudChain} and settle`,
+  exec: async (t, mintAmt: bigint, eudChain: string) => {
+    const {
+      nobleTools,
+      nobleAgoricChannelId,
+      oracleWds,
+      retryUntilCondition,
+      useChain,
+      usdcOnOsmosis,
+      vstorageClient,
+    } = t.context;
+
+    // EUD wallet on the specified chain
+    const eudWallet = await createWallet(
+      useChain(eudChain).chain.bech32_prefix,
+    );
+    const EUD = (await eudWallet.getAccounts())[0].address;
+    t.log(`EUD wallet created: ${EUD}`);
 
-  const recipientAddress = encodeAddressHook(settlementAccount, { EUD });
-  t.log('recipientAddress', recipientAddress);
+    // parameterize agoric address
+    const { settlementAccount } = await vstorageClient.queryData(
+      `published.${contractName}`,
+    );
+    t.log('settlementAccount address', settlementAccount);
 
-  // register forwarding address on noble
-  const txRes = nobleTools.registerForwardingAcct(
-    nobleAgoricChannelId,
-    recipientAddress,
-  );
-  t.is(txRes?.code, 0, 'registered forwarding account');
+    const recipientAddress = encodeAddressHook(settlementAccount, { EUD });
+    t.log('recipientAddress', recipientAddress);
 
-  const { address: userForwardingAddr } = nobleTools.queryForwardingAddress(
-    nobleAgoricChannelId,
-    recipientAddress,
-  );
-  t.log('got forwardingAddress', userForwardingAddr);
-
-  const mintAmount = 800_000n;
-
-  // TODO export CctpTxEvidence type
-  const evidence = harden({
-    blockHash:
-      '0x90d7343e04f8160892e94f02d6a9b9f255663ed0ac34caca98544c8143fee665',
-    blockNumber: 21037663n,
-    txHash: `0xc81bc6105b60a234c7c50ac17816ebcd5561d366df8bf3be59ff3875527617${makeRandomDigits(makeRandomNumber(), 2n)}`,
-    tx: {
-      amount: mintAmount,
-      forwardingAddress: userForwardingAddr,
-    },
-    aux: {
-      forwardingChannel: nobleAgoricChannelId,
+    // register forwarding address on noble
+    const txRes = nobleTools.registerForwardingAcct(
```

Comment:
again, lots going on; my tiny brain wants the roles (user, noble chain, ...) reified as objects.